### ISSUE TITLE
Leaving insert mode teleports cursor to top of file

### ISSUE THREAD

**Issue by jakerr** [vscode-neovim/vscode-neovim#1187]
This happens to me probably 1 in 20 times or so. When I leave insert mode with esc suddenly the cursor jumps to the first character at the top of the file. Pressing ctrl+- to navigate back, takes me back to where I was.

I've already made sure that the process affinity mode is set as per the pinned PSA.

**Comment by justinmk** [comment]
Lots of similar "weird behavior on ESC" issues reported recently:

- https://github.com/vscode-neovim/vscode-neovim/issues/1185
- https://github.com/vscode-neovim/vscode-neovim/issues/1182
- https://github.com/vscode-neovim/vscode-neovim/issues/1178

**Comment by jakerr** [comment]
Yes, sorry I searched open bugs for "top of file" and didn't find any. I did some more searching in closed issues and also found #1012 which has reproduction steps that that describe exactly what I'm seeing.

That bug was closed when the author didn't respond to a request to try with an empty init.vim.

This prompted me to try a binary search on my init.vim settings and it turned out to be caused by:

```.vim
" Insert mode
inoremap <C-j> <ESC>:m .+1<CR>==gi
inoremap <C-k> <ESC>:m .-2<CR>==gi
```

Which is used for ctrl+j, crtl+k line movement ala: https://dockyard.com/blog/2013/09/26/vim-moving-lines-aint-hard
I'm not sure why those <C-j> or <C-k> remapped keys would be triggered or have an effect by just inserting a new line and typing `//` to continue the comment as described in #1012 but it does indeed reproduce under those conditions.

At least I have an easy workaround.
